src/controller/ebike_app.c

```diff
@@ -118,77 +106,89 @@ static void check_brakes(void);
 static void calc_cadence(void);
 static void calc_crank_power(void);
 static void calc_wheel_speed(void);
-static void calc_motor_temperature(void);
-static void calc_filtered_battery_voltage(void);
+static void get_battery_voltage_filtered(void);
+static void get_battery_current_filtered(void);
 
 static void apply_boost();
-static void apply_assist();
+static void apply_power_assist();
+static void apply_torque_assist();
 static void apply_emtb();
-static void apply_cadence_only();
-static void apply_throttle();
+static void apply_cadence_assist();
+static void apply_virtual_throttle();
 static void apply_walk_assist();
 static void apply_cruise();
+static void apply_throttle();
 static void apply_speed_limit();
 static void apply_temperature_limiting();
 
+
+
 void ebike_app_controller (void)
 {
-  calc_filtered_battery_voltage();
+  get_battery_voltage_filtered();
+  get_battery_current_filtered();
+  
   calc_cadence();
   calc_crank_power();
   calc_wheel_speed();
-  calc_motor_temperature();
   
   check_system();
   check_brakes();
   
-  ebike_control_motor();
   communications_controller();
+  ebike_control_motor();
 }
 
 
+
 static void ebike_control_motor (void)
 {
   // reset control variables
   ui8_adc_battery_current_target = 0;
   ui8_duty_cycle_target = 0;
   
   // boost
-  apply_boost();
+  //apply_boost();
   
-  // assist
-  apply_assist();
+  // power assist
+  apply_power_assist();
   
-  // throttle
-  apply_throttle();
+  // torque assist
+  apply_torque_assist();
   
-  // cruise
-  apply_cruise();
+  // eMTB
+  apply_emtb();
```

Comment:
All these functions seem to check the riding mode first.
Wouldn't it be more legible to just have a switch case here checking the riding mode, and then calling the relevant apply_...() function ?

Reply:
Doing a lot of things at the same time. Compare the code to 0.19.0 and you will see what I mean. It will be improved when I have checked my priority items.